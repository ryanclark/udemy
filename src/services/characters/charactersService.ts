import { createApiUrl, fetchRequest } from '../utils';
import { Character } from '../../models/Character';

export const RESULTS_PER_PAGE = 20;

export function getCharacters(offset: number, nameStartsWith?: string) {
  const url = createApiUrl('/characters', { offset, nameStartsWith });

  return fetchRequest<Character[]>(url);
}
