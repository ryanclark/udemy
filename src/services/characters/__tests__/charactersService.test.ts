import * as fetchMock from 'fetch-mock';

import { getCharacters } from '../charactersService';

describe('Characters Service', () => {
  it('should call the characters endpoint correctly', async () => {
    fetchMock
      .restore()
      .getOnce(
        'begin:https://gateway.marvel.com/v1/public/characters',
        {
          status: 200,
          body: {
            code: 200,
            status: 'Ok',
            copyright: '',
            data: {
              offset: 0,
              limit: 0,
              results: [
                {
                  id: 1,
                },
                {
                  id: 2,
                }
              ]
            },
          },
        },
      );

    const result = await getCharacters(0);

    expect(result.data.results.length).toBe(2);
    expect(result.data.results[0].id).toBe(1);

    expect(result).toMatchSnapshot();
  });
});
