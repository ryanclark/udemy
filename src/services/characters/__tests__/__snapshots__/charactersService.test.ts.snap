// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters Service should call the characters endpoint correctly 1`] = `
Object {
  "code": 200,
  "copyright": "",
  "data": Object {
    "limit": 0,
    "offset": 0,
    "results": Array [
      Object {
        "id": 1,
      },
      Object {
        "id": 2,
      },
    ],
  },
  "status": "Ok",
}
`;
